scripts/vendor_browserify.sh

```diff
@@ -1,22 +1,427 @@
+    *) sedi='-i' ;;
+  esac
+
+  sed "${sedi[@]}" "$@"
+}
+
+capture_and_run() {
+  CMDS="${CMDS}# "$(echo "$@" | sed "s#${RULES_NODEJS_DIR}/##g")$'\n'
+  "$@"
+}
+
+readonly RULES_NODEJS_DIR=$(cd $(dirname "$0")/..; pwd)
+readonly THIRD_PARTY_DIR=${RULES_NODEJS_DIR}/third_party
+readonly DATE=$(date)
+readonly PACKAGE_JSON_SHA256=$(cat ${THIRD_PARTY_DIR}/package.json | openssl dgst -sha256)
+readonly YARN_LOCK_SHA256=$(cat ${THIRD_PARTY_DIR}/yarn.lock | openssl dgst -sha256)
+
+readonly BROWSERIFY_PLUGINS=(
+  '@babel/plugin-transform-modules-commonjs'
+)
+
+(
+  cd ${THIRD_PARTY_DIR}
+  yarn install
+
+  ##############################################################################
+  # Vendor browserify
+  ##############################################################################
+  (
+    CMDS=""
+    readonly PKG="browserify"
+    readonly SRC_DIR=${THIRD_PARTY_DIR}/node_modules/${PKG}
+    readonly DST_DIR=${THIRD_PARTY_DIR}/npm/node_modules/${PKG}
+
+    echo "Vendoring ${PKG} to ${DST_DIR}"
+    rm -rf ${DST_DIR}
+    mkdir -p ${DST_DIR}
+
+    echo "- compiling ${PKG} with ncc"
+    capture_and_run ${THIRD_PARTY_DIR}/node_modules/.bin/ncc build ${SRC_DIR}/bin/cmd.js -o ${DST_DIR}
+
+    echo "- local mod: revert https://github.com/browserify/browserify/pull/1801"
+    capture_and_run sedi 's#parent.id !== self._mdeps.top.id#parent.id#' ${DST_DIR}/index.js
+
+    echo "- local mod: workaround https://github.com/zeit/ncc/issues/461"
+    capture_and_run sedi "s#require('process/browser.js')#require('./browser1')#" ${DST_DIR}/main.js
+
+    echo "- copy LICENSE"
+    capture_and_run cp -f ${SRC_DIR}/LICENSE ${DST_DIR}
+
+    echo "- minifying ${PKG} with terser"
+    capture_and_run ${THIRD_PARTY_DIR}/node_modules/.bin/terser --compress --mangle --comments '/(^!|@license|@preserve|Copyright)/' -- ${DST_DIR}/index.js > ${DST_DIR}/index.min.js
+
+    echo """# Generated by /scripts/vendor_npm.sh
+load(\"//internal/js_library:js_library.bzl\", \"js_library\")
+
+package(default_visibility = [\"//visibility:public\"])
+
+licenses([\"notice\"])
+
+# Vendored on ${DATE} from npm with /scripts/vendor_npm.sh
+exports_files([\"LICENSE\"])
+
+js_library(
+    name = \"$(basename ${PKG})\",
+    srcs = glob([\"**/*.js\"]),
+    module_name = \"${PKG}\",
+)
+
+filegroup(
+    name = \"package_contents\",
+    srcs = glob([\"**/*.js\"]) + [
+        \"advanced.txt\",
+        \"BUILD.bazel\",
+        \"LICENSE\",
+        \"usage.txt\",
+    ],
+)
+
+# shell commands:
+${CMDS}
+# package.json:
+$(cat ${THIRD_PARTY_DIR}/package.json | awk '{print "# "$0}')
+
+# yarn.lock
+$(cat ${THIRD_PARTY_DIR}/yarn.lock | awk '{print "# "$0}')
+""" > ${DST_DIR}/BUILD.bazel
+  )
+
+  ##############################################################################
+  # Vendor ieee754
+  # (required by browserify)
+  ##############################################################################
+  (
+    CMDS=""
+    readonly PKG="ieee754"
```

Comment:
it looks like this is intended to loop through the packages, but then the logic is all copy-pasta anyway - do you mean to refactor this script before merging?

Reply:
I'll refactor and cleanup a bit

Reply:
done